web-crawler/src/main/java/com/google/sps/webcrawler/RelevancyChecker.java

```diff
@@ -24,21 +24,34 @@
 import com.google.cloud.language.v1.LanguageServiceClient;
 import com.google.cloud.language.v1.LanguageServiceSettings;
 import com.google.sps.data.NewsArticle;
+import java.io.IOException;
 
 // @TODO [Compute the salience of other features, such as location, election, to improve accuracy.]
-/** 
- * Provides a tool for checking the relevancy of content to a candidate.
+/**
+ * Provides a tool for checking the relevancy of news article content to a candidate.
  */
 public class RelevancyChecker {
   // @TODO [Calculate a meaningful salience threshold.]
-  private static final double salienceThreshold = 0.5;
+  private static final double SALIENCE_THRESHOLD = 0.5;
+  private static LanguageServiceClient languageServiceClient;
+  static {
+    try {
+      languageServiceClient = LanguageServiceClient.create();
+    } catch (IOException e) {
+      languageServiceClient = null;
+    }
+  }
 
   /** 
    * Checks whether the {@code newsArticle} is relevant to the {@code candidateName} of
    * interest. Defines relevancy as the salience of {@code candidateName} in the content,
-   * and defines sufficient relevancy with {@salienceThreshold}.
+   * and defines sufficient relevancy with {@code SALIENCE_THRESHOLD}.
    */
-  public static boolean isRelevant(NewsArticle newsArticle, String candidateName) {
+  public static boolean isRelevant(NewsArticle newsArticle, String candidateName) throws
```

Comment:
I wonder if it'd be better to actually make these non-static, since if it fails to create the `LanguageServiceClient` this whole thing is kind of useless?

You could even have 2 constructors:
- one with 0 params that attempts to create the `LanguageServiceClient` similar to what you have here
- one with a param of `LanguageServiceClient` passed in so that it makes testing easier (i.e. in tests could pass in `MockLanguageServiceClient`)

Reply:
The reason why I made it static and used a static initializer is because I originally intended `RelevancyChecker` to be a toolbox class and all its methods to be static. Since static methods access `languageServiceClient`, it has to be static as well.

Reply:
Would you recommend making `languageServiceClient` and the toolbox method `isRelevant()` non-static? If so, should I also modify the other toolbox classes, like `ContentProcessor`, to make them consistent?

Reply:
Yes, so `languageServiceClient` would be an instance variable, and both `isRelevant()` and `computeSalienceOfName()` would be non-static methods.

For the other toolbox classes, probably no need to make them non-static unless they are inconvenient for testing (i.e. I could see maybe making `NewsContentExtractor` non-static if you wanted to use a mock `HtmlParser` in tests). I mostly mentioned this here since there could be errors upon creation of `LanguageServiceClient`, so you'd probably just want to catch that upfront in the initial class instantiation.

Reply:
For instance, when you use the `RelevancyChecker` in `WebCrawler`, the `isRelevant()` call in `compileNewsArticle()` will always fail if the initial creation of `LanguageServiceClient` failed, so there's no real point to keep trying to call `isRelevant()` over and over again since you know it'll always fail at that point.

Reply:
This makes sense, I agree that this is much more important than whether the toolbox class is static. I'll make the changes